cookie.go

```diff
@@ -119,8 +122,12 @@ func (c *Cookie) SameSite() CookieSameSite {
 }
 
 // SetSameSite sets the cookie's SameSite flag to the given value.
+// set value CookieSameSiteNoneMode will set Secure to true also to avoid browser rejection
 func (c *Cookie) SetSameSite(mode CookieSameSite) {
 	c.sameSite = mode
+	if mode == CookieSameSiteNoneMode {
+		c.SetSecure(true)
+	}
```

Comment:
I don't think we should do this by default. This might cause cookies to work if the server is HTTP only. Requiring Secure for SameSiteNone is a separate flag in Chrome that isn't enabled by default.

Reply:
In the specifications same-site with value "None" must have Secure parameter otherwise the cookie is ignored (https://tools.ietf.org/html/draft-west-cookie-incrementalism-00)
I hesitate between call SetSecure(true) in SetSameSite and AppendBytes() 

cookie.go L280
`if c.secure {`
to
`if c.secure || c.sameSite == CookieSameSiteNoneMode {`

I think we should force the value because the specification make it clear and developers could not be aware.

Reply:
My two cents.
https://tools.ietf.org/html/draft-west-cookie-incrementalism-00#section-3.2 the specification is clear. What we can consider is create an extra flag in `fasthttp` configuration to `refuse non secure SameSite None` cookies. This will affect how we parse cookies, but I am not sure how it should affect writing cookies. It is really easy to force a non secure cookie with SameSite None.

Perhaps we can include a `Valid() error` method on cookie, who can verify the inner state of the cookie, like parse but to itself.

it is better than just refuse `SetSecure(false)` if `SameSite=None` IMHO